tools/dynamo/gb_id_mapping.py

```diff
@@ -0,0 +1,146 @@
+                        if (
+                            isinstance(node.func, ast.Name)
+                            and node.func.id == "unimplemented_v2"
+                        ):
+                            is_unimplemented = True
+                        elif (
+                            isinstance(node.func, ast.Attribute)
+                            and node.func.attr == "unimplemented_v2"
+                        ):
+                            is_unimplemented = True
+
+                        if is_unimplemented:
+                            info = {
+                                "gb_type": None,
+                                "context": None,
+                                "explanation": None,
+                                "hints": [],
+                                "from_exc": None,
+                            }
+
+                            for kw in node.keywords:
+                                if kw.arg in info:
+                                    param_source = get_source_segment(source, kw.value)
+                                    if param_source:
+                                        info[kw.arg] = clean_string(param_source)
+                                    elif isinstance(kw.value, ast.Constant):
+                                        info[kw.arg] = kw.value.value
+                                    else:
+                                        info[kw.arg] = "DYNAMIC_CONTEXT"
+
+                            results.append((file_path, node.lineno, info))
+            except SyntaxError:
+                print(f"Syntax error in {file_path}")
+
+    return results
+
+
+def create_registry():
+    calls = find_unimplemented_v2_calls()
+    registry = {}
+
+    gb_types = {}
+    for file_path, line, info in calls:
+        if info["gb_type"] and info["gb_type"] not in gb_types:
+            gb_types[info["gb_type"]] = info
+
+    for i, (gb_type, info) in enumerate(sorted(gb_types.items()), 1001):
+        gb_id = f"GB{i}"
+
+        hints = info["hints"]
+        if isinstance(hints, str):
+            if hints.startswith("[") and hints.endswith("]"):
+                try:
+                    items = re.findall(r'"([^"]*)"', hints)
+                    if items:
+                        hints = items
+                    elif "*graph_break_hints." in hints:
+                        hints = expand_hints([hints])
+                except:
+                    pass
+
+        registry[gb_id] = {
+            "Version": "v1.0",
+            "Gb_type": clean_string(gb_type),
+            "Context": clean_string(info["context"]),
+            "Explanation": clean_string(info["explanation"]),
+            "Hints": hints if hints else [],
+            "From_exc": info["from_exc"],
+        }
+
+    registry_path = Path(__file__).parent / "graph_break_registry.json"
+    with open(registry_path, "w") as f:
+        json.dump(registry, f, indent=2)
+
+
+if __name__ == "__main__":
```

Comment:
You can use `argparse` to handle custom input (dynamo) directory and json output.